src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketWebhookConsumer.java

```diff
@@ -193,22 +204,15 @@ private void getJobs(RefChangedDetails refChangedDetails, BitbucketWebhookTrigge
                 .forEach(triggerDetails -> triggerDetails.getTrigger().trigger(requestBuilder.build()));
     }
 
-    private void triggerJob(PullRequestWebhookEvent event, RefChangedDetails refChangedDetails) {
+    private void triggerJob(PullRequestWebhookEvent event, RefChangedDetails refChangedDetails,
+                            Optional<BitbucketServerConfiguration> server) {
         try (ACLContext ctx = ACL.as(ACL.SYSTEM)) {
             BitbucketWebhookTriggerRequest.Builder requestBuilder = BitbucketWebhookTriggerRequest.builder();
             event.getActor().ifPresent(requestBuilder::actor);
 
             getJobs(refChangedDetails, requestBuilder);
-            Optional<BitbucketServerConfiguration> server = bitbucketPluginConfiguration.getValidServerList()
-                                                                                        .stream()
-                                                                                        .filter(serverConfig -> refChangedDetails.getRepository()
-                                                                                                                                 .getSelfLink()
-                                                                                                                                 .contains(serverConfig.getBaseUrl()))
-                                                                                        .findFirst();
-            if (server.isPresent()) {
-                //need to do this for all config
-                pullRequestStore.addPullRequest(server.get().getId(), event.getPullRequest());
-            }
+            server.ifPresent(bitbucketServerConfiguration ->
```

Comment:
...and then you can shortern the variable in the lambda to `configuration` etc. Just a suggestion as I didn't immediately equate `server` with `serverConfiguration`